packages/gatsby-plugin-manifest/src/gatsby-node.js

```diff
@@ -20,12 +20,11 @@ function generateIcons(icons, srcIcon) {
 
 exports.onPostBootstrap = (args, pluginOptions) =>
   new Promise((resolve, reject) => {
-    const { icon } = pluginOptions
-    const manifest = { ...pluginOptions }
+    const { icon, ...manifest } = pluginOptions
 
     // Delete options we won't pass to the manifest.webmanifest.
     delete manifest.plugins
```

Comment:
see how `pluginOptions` is destructured - it gets `icon` and rest of it will be `manifest` object (without `icon` field)

`plugins` is something that gatsby automatically passes

Reply:
ah...okay by merging them I didn't realize it remove icon, makes sense.

Reply:
I also added test to `gatsby-node` to make sure `icon`, `plugins` and `legacy` don't get into manifest file :)